tests/vcs/git/GitCommitBearTest.py

```diff
@@ -288,6 +288,24 @@ def test_body_checks(self):
                              body_regex=r'TICKER\s*CLOSE\s+[1-9][0-9]*'), [])
         self.assert_no_msgs()
 
+        # Testing invalid emails
+        self.git_commit('Shortlog\n\n'
+                        'invalid#!format!*!@email.com\n'
+                        'another.*invalid@format\n'
+                        'validformat@email.check\n'
+                        'validdomain@gmail.com\n'
+                        'Fix 2017')
+        message = 'Body contains these invalid emails:\n'
+        message += ' invalid#!format!*!@email.com\n'
+        message += ' another.*invalid@format\n'
+        # Test with email_dns_check set to False
+        self.assertEqual(self.run_uut(), [message])
+        # Test with email_dns_check set to True
+        message += ' validformat@email.check\n'
+        self.assertEqual(self.run_uut(email_dns_check=True), [message])
```

Comment:
With the [unittest.mock](https://docs.python.org/dev/library/unittest.mock.html) module you can temporarily replace certain functions with mocks, so you don't actually have to run them. You can control the output and thus test your code under certain situations. 

In you case you could for example do something lilke this:

``` python
# test successful DNS lookup
with mock.patch("bears/vcs/git/GitCommitBear.is_email", return_value=[]):
   message = "... whatever your expected output is" 
   self.assertEqual(self.run_uut(email_dns_check=True), [message])

# test failing DNS lookup (returning a non-empty list is enough, but you could replace it with an actual error
with mock.patch("bears/vcs/git/GitCommitBear.is_email", return_value=["error"]):
   message = "... whatever your expected output is" 
   self.assertEqual(self.run_uut(email_dns_check=True), [message])

This way you still test all the functionality of the code that you wrote, while not actually going out to the net.